sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/aggregate/linearRegression.scala

```diff
@@ -311,7 +312,8 @@ case class RegrIntercept(left: Expression, right: Expression) extends Declarativ
 
   private val covarPop = new CovPopulation(right, left)
 
-  private val varPop = new VariancePop(right)
+  private val varPop = new VariancePop(If(And(IsNotNull(left), IsNotNull(right)),
```

Comment:
instead of adding null check to the underlying `VariancePop`, shall we add it to `RegrIntercept`? e.g. we can add null check to `updateExpressions`.

Reply:
I tried changing it locally and related test cases passed. It seems that the logic is more complicated. Is it worth updating like this? WDYT @cloud-fan 
```
  override lazy val updateExpressions: Seq[Expression] = {
    val isNull = left.isNull || right.isNull
    val updateResult = covarPop.updateExpressions ++ varPop.updateExpressions
    aggBufferAttributes.zip(updateResult).map { case (oldValue, newValue) =>
      If(isNull, oldValue, newValue)
    }
  }
```

Reply:
I think it's more efficient, as we do null check earlier, before we execute `VariancePop`

Reply:
Actually, the value is calculated regardless of whether it is null with code `val updateResult = covarPop.updateExpressions ++ varPop.updateExpressions`. 

It's similar to the `updateExpressionsDef` func in `Covariance`. The value is calculated first, and finally different values ​​are returned depending on whether it is null.

Reply:
I haven't found a better way to implement this logic. If can use `if else` directly, it may not be calculated in advance, but it cannot be expressed directly here.

Reply:
I updated it, would you like to take another look?  Thank you. @cloud-fan